### PR TITLE
Modal service + Newsletter Sign-up Modal

### LINKED ISSUE #33: Now-u newsletter popup
After approx 30 seconds of being on the website, a popup should be shown to sign up for our newsletter. The form should be implemented using mail chimp (it was in v2). Designs are in XD

### PR DESCRIPTION
**Pull request by ArronAult**
# Description

- Added a ModalService to facilitate the creation of a newsletter sign-up modal which can open a Modal outside of the current component and close a Modal by ref.
- Added a newsletter sign-up modal which opens after 30 seconds on any page on the site. If the user closes the modal without signing up then they won't see the modal again until their next session. If they successfully sign-up then they won't see it again (unless they clear localStorage). On successfully signing-up, the modal with close after 5 seconds.
- Various small changes to facilitate required designs + QOL improvements.

Fixes #33

# Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have reviewed the 'files changed' tab on github to ensure all changes are expected
- [x] I have added someone to review the pr

### CODE REVIEW

File: src/components/modal/Modal.jsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+    this.modalContainer.focus();
+  }
+
+  componentWillUnmount() {
+    this.cleanup();
+  }
+
+  render() {
+    const { isOpen } = this.state;
```
Comment (JElgar) [inline src/components/modal/Modal.jsx:57]: Is this needed? Or can the modal be destroyed (unmount) when it's closed?

File: src/components/modal/Modal.module.scss [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  top: 14px;
+
+  &.light {
+    color: $darkGrey;
+  }
+
+  &.dark {
+    color: $white;
+  }
+}
```
Comment (JElgar) [inline src/components/modal/Modal.module.scss:79]: I would probably do a bit more nesting here - e.g will this .closeButton styling apply elsewhere or is it always gonna be a child of the dialog?

File: src/components/newsletter-signup-modal/NewsletterSignupModal.jsx [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+  }
+
+  /**
+   * The content of the iframe can't be accessed due to cross-origin policy so we can only assume the
+   * subscription was successful for the moment and act on that assumption.
+   *
+   * If the content of the iframe is attempted to be accessed then a DOMException is thrown. This behaviour
+   * is used to be able to determine whether a response has returned from the form.
+   *
+   * TODO: Use API endpoint with fetch() to sign-up users to mailchimp once it is implemented
```
Comment (JElgar) [inline src/components/newsletter-signup-modal/NewsletterSignupModal.jsx:30]: Ah yes @davidmcginn is there an endpoint for this?

File: src/components/newsletter-signup-modal/NewsletterSignupModal.module.scss [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+  }
+}
+
+.subHeader {
+  font-size: 20px;
+
+  @media (min-width: $medium) {
+    font-size: 24px;
+  }
+}
```
Comment (JElgar) [inline src/components/newsletter-signup-modal/NewsletterSignupModal.module.scss:27]: Similar thing with nesting here - eg is subHeader a child of the header (I haven't check so maybe not)?

File: src/domain/faqs/FAQs.module.scss [hunk 5]
```diff
@@ -60,7 +60,7 @@
         text-align: center;
     }
     & p > span{
```
Comment (JElgar) [inline src/domain/faqs/FAQs.module.scss:62]: `& p` is the same as just  `p` in this example ( I know this wasn't you but they still annoy me - there are a few below if you care enough)

Comment (JElgar) [reply src/domain/faqs/FAQs.module.scss:62]:
```suggestion
    p > span{
```

File: src/components/modal/Modal.jsx (hunk 1 above)
Comment (ArronAult) [reply src/components/modal/Modal.jsx:57]: Doing it manually is an anti-pattern in React (afaik). This is the proper way to do it and replicates a parent controlling the rendering of a child through a prop (`handleClose` can be called through the `ModalService`). React deals with the unmounting of it once the component is no longer needing to be rendered.

File: src/domain/faqs/FAQs.module.scss (hunk 5 above)
Comment (ArronAult) [reply src/domain/faqs/FAQs.module.scss:62]: Probably best to do this in a separate change.

File: src/components/newsletter-signup-modal/NewsletterSignupModal.module.scss (hunk 4 above)
Comment (ArronAult) [reply src/components/newsletter-signup-modal/NewsletterSignupModal.module.scss]: `header` and `subHeader` are `Text` components at the same level on the component.

**Pull request merged**